### ISSUE TITLE
bug: when selecting text in a cell after hitting f2, NVDA does not read what is selected when shift+arrows are used.

### ISSUE THREAD

**Issue by marrie** [nvaccess/nvda#15843]
### Steps to reproduce:
1. launch nvda
2. launch excel
3. open a workbook with filled in text or just fill in the current workbook.
4. hit f2 to access the formula editor
5. use control+arrows to review the text
6. do the same with shift to select text.
### Actual behavior:

 I don't hear any text spoken when using shift and arrow keys.
### Expected behavior:

 I should hear text spoken when  I use my selectrion keys similar to chrome, outlook, etc. this includes shift home and end, control shift home and end as well.
#### NVDA installed/portable/running from source:
installed
#### NVDA version:
23.x
#### Windows version:
Windows 11 22H2 (AMD64) build 22621.2715
#### Name and version of other software in use when reproducing the issue:
microsoft 2016 m365  I'm usign excel from said office suite.
#### Does the issue still occur after restarting your computer?
yes.
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, I can regress this to when I noticed this in july 2023.
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

another comment. We have an [active thread](https://groups.io/g/M365-Accessibility/topic/when_editing_a_cell_in_excel/102785139?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,0,102785139,previd%3D9223372036854775807,nextid%3D1700079659683808154&previd=9223372036854775807&nextid=1700079659683808154) explaining what is going on and a workaround. I want to see if this is also a microsoft issue. Thanks.

**Comment by Qchristensen** [comment]
(Updated original description / title just to fix a couple of typos which may impact easy reading of the text).

TL;DR: I can repro with Excel's "Allow editing directly in cells" enabled, and NVDA's use UIA in Excel disabled.


I tested this with the following setup:
NVDA 2023.3
Windows 11 (64-bit) Version: 22H2, Build: 22621.2715
Office 365 (64-bit) Version: 16.0.17029.20028 

What I found was:
With "Allow editing directly in cells" ENABLED in Excel's advanced options and NVDA's "Use UI Automation to access MS Excel spreadsheet controls when available" also enabled:

1)  When I press F2 to edit a cell, then use SHIFT+ARROWS to select text, NVDA reads each letter as it is selected "T selected, H selected, E selected" if I select the word THE one letter at a time for instance).

2)  If I disable NVDA's Use UIA in Excel option, letters are NOT read as I select them.

3)  Leaving NVDA's Use UIA in Excel option disabled and disabling "Allow editing directly in cells" in Excel, NVDA reads letters as they are selected.

4)  If I then enable Use UIA in Excel in NVDA, letters are read as they are selected.

So, in short, the only situation where letters are NOT read as they are selected for me is with NVDA's Use UIA in Excel disabled AND Excel's allow editing directly in cells enabled.

**Comment by marrie** [comment]
Thanks. Sorry about the typos. I saw those when it was too late.   What are next steps going from here?

**Comment by britechguy** [comment]
Sarah,

Sean Budd marked this as triaged, priority two, app/microsoft-office about a half an hour before your message.  Since I had not commented before now, I hadn't seen anything up to now, but I thought you would have received an email message regarding that status change since you were the issue originator.  It also shows on the GitHub page, which is where I'm typing from, but I'm not 100% positive whether you get an email when such designations are made, but I could have sworn I did in the past.

**Comment by marrie** [comment]
Thanks. No I did not. I normally get an update when things are closed. I will keep an eye on this. Thanks for the update.

**Comment by Adriani90** [comment]
@seanbudd this is very related to #14122. I think NVDA cannot fetch selections and deletions in the edit field in Excel when editing directly in cells is enabled. Should we report this to Microsoft? Or is there anything NVDA can do ybout it?

**Comment by seanbudd** [comment]
Reporting it to Microsoft may be worthwhile, either of us would need to do a technical investigation before determining fault. MS generally has more resources
